python/ray/data/dataset.py

```diff
@@ -747,16 +755,41 @@ def add_column(
                 ray (e.g., num_gpus=1 to request GPUs for the map tasks).
         """
 
-        def add_column(batch: "pandas.DataFrame") -> "pandas.DataFrame":
-            batch.loc[:, col] = fn(batch)
-            return batch
+        def add_column(batch: "pyarrow.Table") -> "pyarrow.Table":
```

Comment:
the typing here is off - batch is DataBatch type right? for example if it is pandas

Reply:
Thanks!